Core/Core/Common/CommonModels/AppEnvironment/AppEnvironmentOverride.swift

```diff
@@ -127,6 +127,26 @@ extension AppEnvironment {
 
 // MARK: - Utils
 
+public struct RoutePath {
```

Comment:
My feeling is that this `RoutePath` struct gives the solution a twist but the abstraction itself provides no added value as opposed to using strings directly. I would move the logic from here to a string extension and use that directly:
```
public extension String {
    func asRoute(in env: AppEnvironment) -> String {
        guard env.api.baseURL != AppEnvironment.shared.api.baseURL else {
            return value
        }
        return env.api.baseURL.appending(path: value).absoluteString
    }
}
```
This way no modification would be needed in the router.